llarp/crypto/crypto.hpp

```diff
@@ -23,36 +23,30 @@ namespace llarp
 
     /// xchacha symmetric cipher
     bool
-    xchacha20(uint8_t*, size_t size, const SharedSecret&, const TunnelNonce&);
+    xchacha20(uint8_t*, size_t size, const SharedSecret&, const SymmNonce&);
```

Comment:
Minor:
So since these nonce types are declared like this:
```C++
using TunnelNonce = AlignedBuffer<TUNNONCESIZE>;
using SymmNonce = AlignedBuffer<NONCESIZE>;
```
I feel like it could be a lot clearer if we did this:
```C++
inline constexpr uint32_t TUNNEL{32}; // this is TUNNONCESIZE
inline constexpr uint32_t SYMMETRIC{24}; // this is NONCESIZE

template <typename sz>
using Nonce = AlignedBuffer<sz>; // or we could add nonce-specfic methods and make this a struct
```
Then the type would be `Nonce<TUNNEL>` or `Nonce<SYMMETRIC>`. Not sure if this might cause issues with different types that have same sizes, but maybe we can figure out a better solution to the needless typedefing haha

Reply:
TunnelNonce isn't used any more anyway, but removing it cascaded into a lot of compiler whack-a-mole on code which is pending changes anyway,  so I was leaving this for until then (at which time we can just have one `Nonce` type, which should have been the case all along it seems).  I can include that change here, but I didn't follow the whack-a-mole rabbit hole to its conclusion so I'm not 100% sure where it terminates (might be really shallow and I ditched it early...)